src/inference/dev_api/openvino/runtime/threading/istreams_executor.hpp

```diff
@@ -166,12 +170,47 @@ class OPENVINO_RUNTIME_API IStreamsExecutor : virtual public ITaskExecutor {
      */
     virtual int get_numa_node_id() = 0;
 
+    /**
+     * @brief Return the id of current socket
+     * @return `ID` of current socket, or throws exceptions if called not from stream thread
+     */
+    virtual int get_socket_id() = 0;
+
     /**
      * @brief Execute the task in the current thread using streams executor configuration and constraints
      * @param task A task to start
      */
     virtual void execute(Task task) = 0;
 };
 
+enum StreamCreateType {
+    STREAM_WITHOUT_PARAM = 0,  // new task_arena with no parameters, no threads binding
+    STREAM_WITH_CORE_TYPE,     // new task_arena with core type, threads binding with core type
+    STREAM_WITH_NUMA_ID,       // new task_arena with numa node id, threads bingding with numa node id
+    STREAM_WITH_OBSERVE        // new task_arena with no parameters, threads binding with observe
+};
+
+/**
+ * @brief      Get current stream information
+ * @param[in]  stream_id stream id
+ * @param[in]  cpu_reservation cpu reservation
+ * @param[in]  org_proc_type_table available processors in the platform
+ * @param[in]  streams_info_table streams information table
+ * @param[in]  stream_numa_node_ids numa_node_ids sorted by stream id
+ * @param[in,out]  stream_nums the list of number of streams which is used to access streams_info_table
+ * @param[out]  stream_type stream create type
+ * @param[out]  concurrency the number of threads created at the same time
+ * @param[out]  core_type core type
+ * @param[out]  numa_node_id numa node id
+ */
+void get_cur_stream_info(const int stream_id,
```

Comment:
The function is exposed as stream executor interface, but used only inside it? Does it make sense to make it private?

Reply:
This function is used for testing. Although it is used only inside, I think it can not be made private.